modules/network/src/main/java/org/apache/ignite/internal/network/DefaultMessagingService.java

```diff
@@ -764,4 +782,116 @@ private InetSocketAddress getFromCacheOrCreateResolved(ClusterNode recipientNode
 
         return address != null ? address : createResolved(recipientNode.address());
     }
+
+    /**
+     * Timeout object worker.
+     */
+    private static class TimeoutWorker extends IgniteWorker {
```

Comment:
Duplicate code. As mentioned above, we can put this into core module and reuse.

Reply:
I moved the code in one palce.